### PR TITLE
CASMINST-5301: CMS, HMS release notes

### PR DESCRIPTION
**Pull request by mharding-hpe**
# Description

* Updates to the CSM release notes based on feedback from CMS and HMS teams.
* Removes differences page, as discussed in previous standup meetings.
* Creates page listing all currently deprecated (but not yet removed) features

# Checklist Before Merging

- [X] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [X] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [X] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: RELEASE_NOTES.md
Comment (mharding-hpe) [inline RELEASE_NOTES.md:25]:
```suggestion
* Added support for using SCSD to enable or disable TPM BIOS setting on Gigabyte and HPE hardware
```

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [inline introduction/deprecated_features/README.md:24]:
```suggestion
```

File: RELEASE_NOTES.md
Comment (mharding-hpe) [inline RELEASE_NOTES.md:30]:
```suggestion
  * NCN and Kubernetes health checks now execute in parallel and eliminate lengthy output for tests that pass
```

**Comment by mharding-hpe** [comment]
Need to get my CMS and HMS comrades to review this

**Review by rfrost-hpe — APPROVED** [review]

**Comment by mharding-hpe** [comment]
This should not merge without approvals from @nieuwsma , @jsollom-hpe , @dlaine-hpe, and @rsjostrand-hpe . (and @rfrost-hpe and @schooler-hpe , but they've already done so)

File: RELEASE_NOTES.md
Comment (rsjostrand-hpe) [inline RELEASE_NOTES.md:37]: @nieuwsma do we need to mention something about the NID allocation issue for this blade type?

File: introduction/deprecated_features/README.md
Comment (rsjostrand-hpe) [inline introduction/deprecated_features/README.md:9]: @nieuwsma should `CSM 1.5` be aligned with to one of our non-optional releases like CSM 1.6?

Comment (mharding-hpe) [reply introduction/deprecated_features/README.md:9]: Definitely it seems like it makes more sense to target the removal for a non-optional release.

Comment (denniswalker) [reply introduction/deprecated_features/README.md:9]: Are there other consumers of this API outside of CSM?

Comment (nieuwsma) [reply introduction/deprecated_features/README.md:9]: potentially.

File: RELEASE_NOTES.md
Comment (nieuwsma) [reply RELEASE_NOTES.md:37]: Um... I dont know. 
I think thats probably something else we handle later.

**Review by nieuwsma — APPROVED** [review]

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [reply introduction/deprecated_features/README.md:9]: Let me know what the consensus is here and I will edit accordingly.

**Review by rsjostrand-hpe — APPROVED** [review]

**Comment by mharding-hpe** [comment]
Waiting on the decision in the outstanding review discussion. Also still need reviews from @jsollom-hpe and @dlaine-hpe

**Review by dlaine-hpe — APPROVED** [review]

**Review by jsollom-hpe — CHANGES_REQUESTED** [review]

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [reply introduction/deprecated_features/README.md:9]: Looking through the page, I see a bunch of acronyms used without ever being defined. CRUS, IMS, SCSD, UAS, UAI, etc. I think it may make more sense for me to make them all into links to our glossary page, to reduce clutter on this page.

Comment (mharding-hpe) [reply introduction/deprecated_features/README.md]: I made a quick pass at doing this. Some of the acronyms did not currently have entries in the glossary. If I have time, I'll add them and link them.

File: RELEASE_NOTES.md
Comment (mharding-hpe) [reply RELEASE_NOTES.md]: Done. Let me know what you think.

**Comment by mharding-hpe** [comment]
Waiting on two things before this can merge:

1. A decision from @denniswalker and @nieuwsma about this discussion: https://github.com/Cray-HPE/docs-csm/pull/2431/#discussion_r966129674

2. Re-review by @jsollom-hpe to make sure that his comments have been adequately addressed.

File: introduction/deprecated_features/README.md
Comment (jsollom-hpe) [inline introduction/deprecated_features/README.md:10]: Please add:
  * The release when BOS v1 will be removed has not been specified yet. However, customers should consider  a migration path from BOS v1 to v2.

**Review by jsollom-hpe — CHANGES_REQUESTED** [review]

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [reply introduction/deprecated_features/README.md:15]: It occurs to me that this statement is true for all deprecated features that haven't yet had their final CSM versions announced. So I added a paragraph at the top of the page which covers this. Please let me know if it covers your concern.

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [reply introduction/deprecated_features/README.md]: Ok -- let me know what version I should specify here.

Comment (rfrost-hpe) [reply introduction/deprecated_features/README.md]: since the only recipe that is coming after 1.3 is 1.6 (Alice) I would assume that is the one

Comment (mharding-hpe) [reply introduction/deprecated_features/README.md]: Ok, I'll make that change. Just need re-review from Jason S now for this to be ready to merge.

File: introduction/deprecated_features/README.md
Comment (mharding-hpe) [inline introduction/deprecated_features/README.md:22]:
```suggestion
  * The v1 HMNFD APIs are targeted for removal in the CSM 1.6 release.
```

**Comment by mharding-hpe** [comment]
Now only need re-review from @denniswalker and @jsollom-hpe to make sure that their review comments have been adequately addressed.

**Review by nieuwsma — APPROVED** [review]

**Review by jsollom-hpe — APPROVED** [review]

**Pull request merged**
